testing/citest/tests/kube_v2_artifact_test.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration test to see if the image promotion process is working for the
+Spinnaker Kubernetes V2 integration.
+"""
+
+# Standard python modules.
+import sys
+import random
+import string
+
+# citest modules.
+import citest.kube_testing as kube
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+import spinnaker_testing.frigga as frigga
+import citest.base
+
+ov_factory = jc.ObservationPredicateFactory()
+
+class KubeV2ArtifactTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the kube v2 artifact test.
+  """
+
+  @classmethod
+  def new_agent(cls, bindings):
+    """Implements citest.service_testing.AgentTestScenario.new_agent."""
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 180
+    return agent
+
+  @classmethod
+  def initArgumentParser(cls, parser, defaults=None):
+    """Initialize command line argument parser.
+
+    Args:
+      parser: argparse.ArgumentParser
+    """
+    super(KubeV2ArtifactTestScenario, cls).initArgumentParser(
+        parser, defaults=defaults)
+
+    defaults = defaults or {}
+    parser.add_argument(
+      '--test_namespace', default='default',
+      help='The namespace to manage within the tests.')
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    """
+    super(KubeV2ArtifactTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    # We'll call out the app name because it is widely used
+    # because it scopes the context of our activities.
+    # pylint: disable=invalid-name
+    self.TEST_APP = bindings['TEST_APP']
+
+    # Take just the first if there are multiple
```

Comment:
Does it matter which namespace this deploys to? Are there problems when multiple tests are targeting the same namespace?

Reply:
Doesn't matter, we make sure all the names are unique enough